### ISSUE TITLE
[UWP] TapGestureRecognizer No Longer Blocks List View Item Tapped Event

### ISSUE THREAD

**Issue by MattRames** [xamarin/Xamarin.Forms#1406]
### Description
TapGestureRecognizers inside the view cell of a list view no longer block the item tapped/item selected event of a list view.

### Steps to Reproduce

1.  Create a list view with item tapped and or item selected events.
2.  Have a control with a tap gesture recognizer inside the view cell.
3.  Tap gesture recognizer doesn't block item tapped / item selected events.

### Expected Behavior

The TapGestureRecognizer should block the item tapped / item selected event from firing. This use to be the behavior.

### Actual Behavior

Both the TapGestureRecognizer and the item tapped / item selected events fire. 

### Basic Information

- Version with issue: 2.5.0.121934
- Last known good version: 2.3.4.270
- IDE: VS 2017 15.5.1
- Platform Target Frameworks:
  - UWP:  15063
- Nuget Packages: Xamarin Forms
- Affected Devices: UWP

### Reproduction Link

Here is a reproduction solution
https://drive.google.com/file/d/1iud-OgALmeTw1Jf1q6xKMGG00v2xsNhF/view?usp=sharing

**Comment by r-a-y-w** [comment]
Same as https://bugzilla.xamarin.com/show_bug.cgi?id=32842 I guess?

**Comment by cschwarz** [comment]
PR which may introduced the change:

https://github.com/xamarin/Xamarin.Forms/pull/842

**Comment by cschwarz** [comment]
Any chance to get this fixed in the next release?

This is blocking us from upgrading from 2.3 to 2.5 and using .NET Standard.

**Comment by MattRames** [comment]
From this https://github.com/xamarin/Xamarin.Forms/pull/1157 it looks like there is a legacy mode that will allow the old behavior.

MyList.On<Windows>().SetSelectionMode(ListViewSelectionMode.Inaccessible);

This bug can more than likely be closed.

**Comment by cschwarz** [comment]
Have you tested it with the reproduction sample?

It still logs:
Item Selected Fired.
Tap Gesture Command Fired.

**Comment by MattRames** [comment]
It's workable in my case because I'm just using ItemTapped to execute a command. 

```
private void ItemsControl_ItemTapped(object sender, ItemTappedEventArgs e)
{
    if (e.Item == null) return;
    if (ItemTappedCommand != null)
    {
        if (ItemTappedCommand.CanExecute(e.Item))
        {
            ItemTappedCommand.Execute(e.Item);
        }
    }
    if (HideDefaultSelection)
    {
        ((ListView)sender).SelectedItem = null;
    }
}
```

**Comment by tchadwick** [comment]
The problem is, in iOS and Android we respond to on list item taps as well as individual buttons on the list items.  So without re-writting our iOS and Android apps, we need individual tap gestures firing before the item selected.

**Comment by samhouts** [comment]
Events fire in this order:

**Comment by JCookVF** [comment]
Any update on this? I've upgraded xamarin to the most recent releases and this is still a big issue for us. Would love to see a fix or at least a work around if anyone knows a good one.

**Comment by mnidhk** [comment]
Indeed, a fix or workaround would be nice!

**Comment by JCookVF** [comment]
At the moment we are working around this by adding a platform specific transparent button on top of the Image that has the same command as the tap gesture recognizer. This way the button gets pressed instead of the image and the button does not trigger a list view item selected event. If you're keeping the command on the gesture recognizer though it is important that the button gets placed on top so that it doesn't trigger the command twice (once for the button, once for the gesture recognizer). Aside from the extra work and code the only downside of this I have seen would be the animation of the button hover/press will now be visible (if you care about that). If that is a problem I think it can be removed via a custom renderer or effect though. I'll add a quick snip of my xaml where I've done this if you want a visual..

![image](https://user-images.githubusercontent.com/41007168/64030478-ad5f5000-cb14-11e9-901a-50c9c133b838.png)

I can't remember the specifics for each platform but I'm pretty sure that placing a control programmatically after another control, like the Platform Specific tag in this case, will draw the control last on the screen? i.e. button on top of the image.

**Comment by mnidhk** [comment]
> At the moment we are working around this by adding a platform specific transparent button on top of the Image that has the same command as the tap gesture recognizer. This way the button gets pressed instead of the image and the button does not trigger a list view item selected event. If you're keeping the command on the gesture recognizer though it is important that the button gets placed on top so that it doesn't trigger the command twice (once for the button, once for the gesture recognizer). Aside from the extra work and code the only downside of this I have seen would be the animation of the button hover/press will now be visible (if you care about that). If that is a problem I think it can be removed via a custom renderer or effect though. I'll add a quick snip of my xaml where I've done this if you want a visual..
> 
> ![image](https://user-images.githubusercontent.com/41007168/64030478-ad5f5000-cb14-11e9-901a-50c9c133b838.png)
> 
> I can't remember the specifics for each platform but I'm pretty sure that placing a control programmatically after another control, like the Platform Specific tag in this case, will draw the control last on the screen? i.e. button on top of the image.

Thanks for your example and writing this here! I'm going to try it!

**Comment by samhouts** [comment]
This issue doesn't seem to have had any activity in a long time. We're working on prioritizing issues and resolving them as quickly as we can. To help us get through the list, we would appreciate an update from you to let us know if this is still affecting you on the latest version of Xamarin.Forms, since it's possible that we may have resolved this as part of another related or duplicate issue. If we don't see any new activity on this issue in the next 30 days, we'll evaluate whether this issue should be closed. Thank you!

**Comment by JCookVF** [comment]
@samhouts Yes this is still an issue in the most recent Xamarin version. I am attaching a repo here that shows a reproduction of the bug.

https://github.com/JCookVF/UWPButtonOnListViewBugExample.git
